### ISSUE TITLE
$removeparam breaks google search suggestions

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/CoreLibs#1464]
AdGuard 7.7 nightly 2 (build 3592, CL 1.7.227)

The rules, which break the site(disabled for apps in filters):
```
||google.*/search$removeparam=ei
||google.*/search$removeparam=gs_lcp
```

Steps to reproduce:
1. Add the rule above;
2. Search anything(better with cyrillic);
3. Click on the search field and on suggested search request

Sample [link](https://www.google.com/search?q=%D0%BD%D0%BE%D0%B3%D0%B8+%D1%88%D0%BF%D0%BE%D1%80%D0%B0&client=ms-android-sonymobile&prmd=ivn&source=lnms&tbm=isch&sa=X&ved=2ahUKEwjo8eGC0ZrxAhXpxIsKHU6dCDQQ_AUoAXoECAIQAQ&cshid=1623794295851&biw=360&bih=512&dpr=2)

<details><summary>Screenshot:</summary>

![image](https://user-images.githubusercontent.com/8361299/121691451-ad54f800-cacf-11eb-821a-78b1b54fcd9d.png)

</details>

Result:
![image](https://user-images.githubusercontent.com/8361299/121691517-c1005e80-cacf-11eb-87e2-da0dbebcf74d.png)

Not reproduced with browser extension.

**Comment by zloyden** [comment]
Another steps to reproduce.
<details><summary>Video:</summary>

https://user-images.githubusercontent.com/33488229/122198924-3f7b4880-cea2-11eb-904b-3fcba8c945d1.mp4
</details><br/>

**Comment by AdamWr** [comment]
It seems that it also breaks links on `baidu.com`.

Steps to reproduce:
1. Add this rule:
`||www.baidu.com/link?url=$removeparam=eqid`
2. Go here -  [https://www.baidu.com/s?ie=utf-8&f=8&rsv_bp=1&rsv_idx=1&tn=baidu&wd=%E7%BE%8E%E5%AE%B9&fenlei=256&rsv_pq=ec8ba36d000af282&rsv_t=ac79%2F5Vm3XZrJdbxRrXSFLmjkBcEfneYl1m66%2BAmhIWs%2FXXvAvuN%2BpBQ%2B%2FY&rqlang=cn&rsv_enter=1&rsv_dl=tb&rsv_sug3=9&rsv_sug1=3&rsv_sug7=101&rsv_sug2=0&rsv_btype=i&prefixsug=%25E7%25BE%258E%25E5%25AE%25B9&rsp=5&inputT=3552&rsv_sug4=4593](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww.baidu.com%2Fs%3Fie%3Dutf-8%26f%3D8%26rsv_bp%3D1%26rsv_idx%3D1%26tn%3Dbaidu%26wd%3D%25E7%25BE%258E%25E5%25AE%25B9%26fenlei%3D256%26rsv_pq%3Dec8ba36d000af282%26rsv_t%3Dac79%252F5Vm3XZrJdbxRrXSFLmjkBcEfneYl1m66%252BAmhIWs%252FXXvAvuN%252BpBQ%252B%252FY%26rqlang%3Dcn%26rsv_enter%3D1%26rsv_dl%3Dtb%26rsv_sug3%3D9%26rsv_sug1%3D3%26rsv_sug7%3D101%26rsv_sug2%3D0%26rsv_btype%3Di%26prefixsug%3D%2525E7%2525BE%25258E%2525E5%2525AE%2525B9%26rsp%3D5%26inputT%3D3552%26rsv_sug4%3D4593)
3. Click on a link which redirects to `https://www.120come.com/` (screenshot below)
<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/29142494/124480262-5deca980-dda7-11eb-9ba2-9892383dc1e9.png)

</details>

It should redirects to `https://www.120come.com/`, but instead, it shows message "Not Found".
<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/29142494/124480435-8a082a80-dda7-11eb-98b5-83dd199f01f8.png)

</details>

**Comment by ngorskikh** [comment]
@Alex-302 @AdamWr I can't repro 😢

**Comment by AdamWr** [comment]
I have checked it out with AdGuard for Windows 7.7 nightly 5 (build 3605, CL 1.7.234) and with Chrome/Firefox, but it seems that the same happens on MacOS.
With which AdGuard version did you check it out?

By the way, AdGuard extension must be disabled, otherwise this issue will not occur.

**Comment by Alex-302** [comment]
Als easyly reproduced on android.
Try to add all removeparam rules:
```
||www.google.com/url?$removeparam=ved
||google.*/search$removeparam=ved
||google.*/search$removeparam=biw
||google.*/search$removeparam=bih
||google.*/search$removeparam=dpr
||google.*/search$removeparam=sa
||google.*/search$removeparam=source
||google.*/search$removeparam=aqs
||google.*/search$removeparam=sourceid
||google.*/search$removeparam=ei
||google.*/search$removeparam=gs_lcp
```

**Comment by ameshkov** [comment]
@ngorskikh are you testing with CL 1.8?

**Comment by ngorskikh** [comment]
Guys, I could not reproduce this, no matter how hard I've tried. As I said earlier, this issue might be due to a bug with percent-encoding in 1.7, which has been fixed in 1.8. Could you try one of the latest nightlies that shipped with 1.8?

**Comment by AdamWr** [comment]
I can't reproduce these issues with the latest nightly build (Version 7.7 nightly 10 (build 3626, CL 1.8.141)).
